### PR TITLE
Update ingress controller, and fix integration test

### PR DESCRIPTION
**Pull request by kairen**
In the recent observation, I found the ingress integration test failed in some environment, I think the reason that test fails is that ingress pod not ready yet, so this commit use `Retry()` to make sure that the test not only run once. 

In addition update image for Ingress controller.

/cc @r2d4 @dlorenc

### CODE REVIEW

**Comment by dlorenc** [comment]
@minikube-bot ok to test

File: pkg/minikube/constants/constants.go [hunk 1]
```diff
@@ -170,6 +170,11 @@ var LocalkubeCachedImages = []string{
 	// Dashboard
 	"gcr.io/google_containers/kubernetes-dashboard-amd64:v1.6.3",
 
+	// Ingress Controller
```
Comment (dlorenc) [inline pkg/minikube/constants/constants.go]: These images get cached on all systems - I don't think we want to add them here.

@r2d4 where do you think we should add these? Maybe in common.sh using the `minikube cache` command in a list?

Comment (kairen) [reply pkg/minikube/constants/constants.go]: Hi @r2d4 Could you please help to check this?

**Comment by kairen** [comment]
Hi @dlorenc, I revert cache because of the cache image should not add them here.

**Comment by dlorenc** [comment]
Thanks!

**Pull request merged**